### PR TITLE
Make use of pathlib in more places

### PR DESCRIPTION
**Pull request by tysmith**

### CODE REVIEW

File: grizzly/common/reporter.py [hunk 1]
```diff
@@ -373,15 +366,15 @@ def _process_rr_trace(self, report):
             LOG.info("rr trace exists at %r", s3_url)
             self.add_extra_metadata("rr-trace", s3_url)
             # remove traces so they are not reported to FM
-            rmtree(trace_path)
+            rmtree(str(trace_path))
```
Comment (pyoor) [inline grizzly/common/reporter.py:369]: No need to cast to string as rmtree accepts Path objects.

File: grizzly/common/storage.py [hunk 2]
```diff
@@ -445,25 +444,24 @@ def load_single(cls, path, adjacent=False, load_assets=True, copy=True):
...
             asset_path = info.get("assets_path", None)
-            for dpath, _, files in walk(entry_point.parent):
-                dpath = Path(dpath)
+            for entry in Path(entry_point.parent).rglob("*"):
+                if not entry.is_file():
+                    continue
+                location = entry.relative_to(entry_point.parent).as_posix()
                 # ignore asset path
-                if asset_path and dpath.samefile(entry_point.parent / asset_path):
+                if asset_path and location.startswith(asset_path):
```
Comment (pyoor) [inline grizzly/common/storage.py:452]: Minor nit - you could also use samefile here.

File: grizzly/common/report.py [hunk 3]
```diff
@@ -106,8 +104,8 @@ def cleanup(self):
         Returns:
             None
         """
-        if self.path and isdir(self.path):
-            rmtree(self.path)
+        if self.path and self.path.is_dir():
+            rmtree(str(self.path))
```
Comment (pyoor) [inline grizzly/common/report.py:108]: No need to cast to string as rmtree accepts Path objects.

File: grizzly/common/reporter.py [hunk 4]
```diff
@@ -222,13 +218,13 @@ def _pre_submit(self, report):
...
-        trace_path = pathjoin(report.path, "rr-traces")
-        if isdir(trace_path):
+        trace_path = report.path / "rr-traces"
+        if trace_path.is_dir():
             LOG.info("Ignored rr trace")
             self.add_extra_metadata("rr-trace", "ignored")
             # remove traces so they are not uploaded to FM (because they are huge)
             # use S3FuzzManagerReporter instead
-            rmtree(trace_path)
+            rmtree(str(trace_path))
```
Comment (pyoor) [inline grizzly/common/reporter.py:227]: No need to cast to string as rmtree accepts Path objects.

File: grizzly/common/reporter.py [hunk 5]
```diff
@@ -335,22 +328,22 @@ class S3FuzzManagerReporter(FuzzManagerReporter):
...
+        latest_trace = (src / "latest-trace").resolve(strict=True)
+        assert latest_trace.is_dir(), "missing latest-trace directory"
+        rr_arc = dest / "rr.tar.bz2"
         LOG.debug("creating %r from %r", rr_arc, latest_trace)
         with tar_open(rr_arc, "w:bz2") as arc_fp:
-            arc_fp.add(latest_trace, arcname=basename(latest_trace))
+            arc_fp.add(str(latest_trace), arcname=latest_trace.name)
         # remove path containing uncompressed traces
-        rmtree(src)
+        rmtree(str(src))
```
Comment (pyoor) [inline grizzly/common/reporter.py:338]: No need to cast to string as rmtree accepts Path objects.

**Review by pyoor — APPROVED** [review]
Just a few minor nits, otherwise it looks good.

File: grizzly/common/storage.py (hunk 2 above)
Comment (pyoor) [reply grizzly/common/storage.py:452]: @tysmith shared this over matrix - [Shutil only supports path objects in 3.9](https://docs.python.org/3/library/shutil.html#shutil.move)

Comment (tysmith) [reply grizzly/common/storage.py:452]: This is checking if a file is in `asset_path` not if it is `asset_path`

**Pull request merged**